metron-interface/metron-alerts/src/app/alerts/alerts-list/table-view/table-view.component.ts

```diff
@@ -259,19 +259,25 @@ export class TableViewComponent implements OnInit, OnChanges, OnDestroy {
   }
 
   deleteOneAlertFromMetaAlert($event, alert: Alert, metaAlertIndex: number) {
-    this.metronDialogBox.showConfirmationMessage('Do you wish to remove the alert from the meta alert?').subscribe(response => {
-      if (response) {
+    let confirmedSubscription = this.dialogService.confirm('Do you wish to remove the alert from the meta alert?').subscribe(r => {
+      if (r === 'Confirmed') {
         this.doDeleteOneAlertFromMetaAlert(alert, metaAlertIndex);
       }
+      if (r !== 'Initial') {
```

Comment:
What if we remove the if statement and simply unsubscribe at the end of this function?
With that, I think we could be able to remove 'Initial' value from ConfirmationType and the dispatching of 'Initial' from the first line of DialogService.confirm function. Could we keep the same functionality with this simplification?

Also, we can call variable 'r' like 'action' if we remove the coco egg.